pyspark/bigdl/keras1/converter.py

```diff
@@ -403,16 +408,218 @@ def create_reshape(self, klayer, kclayer):
+            blayer = BLayer.JoinTable(
+                dimension=klayer.concat_axis,
+                n_input_dims=len(input_shape[0]) - 1,
+                bigdl_type="float")
+        elif klayer.mode == "sum":
+            blayer = BLayer.CAddTable(
+                inplace=False,
+                bigdl_type="float")
+        elif klayer.mode == "mul":
+            blayer = BLayer.CMulTable(bigdl_type="float")
+        elif klayer.mode == "max":
+            blayer = BLayer.CMaxTable(bigdl_type="float")
+        # error in dot
+        elif klayer.mode == "dot":
+            blayer = BLayer.DotProduct(bigdl_type="float")
+        # elif klayer.mode == "ave":
+        #     blayer = BLayer.Mean(dimension=1,
+        #          n_input_dims=-1,
+        #          squeeze=True,
+        #          bigdl_type="float")
+        elif klayer.mode in ['ave', 'cos']:
+            raise Exception("Merge mode `%s` not supported for now" % klayer.mode)  # TODO: whether to support cosine and average
+        else:  # invalid mode or lambda functions
+            raise Exception("Invalid merge mode: %s " % klayer.mode)
         return blayer
 
+    # Advanced Activation Layers
+    # SReLU not supported for now.
+    def create_elu(self, klayer, kclayer):
+        return BLayer.ELU(alpha=float(klayer.alpha),
+                          inplace=False,
+                          bigdl_type="float")
+
+    def create_prelu(self, klayer, kclayer):
+        return BLayer.PReLU(n_output_plane=0,
+                            bigdl_type="float")
+
+    def create_leakyrelu(self, klayer, kclayer):
+        return BLayer.LeakyReLU(negval=float(klayer.alpha),
+                                inplace=False,
+                                bigdl_type="float")
+
+    def create_parametricsoftplus(self, klayer, kclayer):
+        alpha = float(klayer.alpha_init)
+        beta = float(klayer.beta_init)
+        if round(alpha * beta, 4) == 1.0:
+            return BLayer.SoftPlus(beta=beta,
+                                   bigdl_type="float")
+        else:
+            raise Exception("Only alpha_init = 1/beta_init is supported for now")
+
+    def create_thresholdedrelu(self, klayer, kclayer):
+        return BLayer.Threshold(th=float(klayer.theta),
+                                v=0.0,
+                                ip=False,
+                                bigdl_type="float")
+
+    # ZeroPadding layers
+    def __generate_zeropadding1d(self, pad_top, pad_bottom):
+        return BLayer.SpatialZeroPadding(pad_left=0,
+                                         pad_right=0,
+                                         pad_top=pad_top,
+                                         pad_bottom=pad_bottom,
+                                         bigdl_type="float")
+
+    def create_zeropadding1d(self, klayer, kclayer):
+        padding = klayer.padding
+        if isinstance(padding, int):  # int
```

Comment:
try to read those padding value from kclayer to get rid of those checking

Reply:
add unit test for those diff scenario.

Reply:
I have already added unit test for each different input

Reply:
The checking I do here is similar to the code in keras. You mean I can directly get the padding value from kclayer? What type?